rustfmt-core/rustfmt-config/src/config_type.rs

```diff
@@ -280,16 +296,88 @@ macro_rules! create_config {
                 )+
             }
 
+            fn set_width_heuristics(&mut self, heuristics: WidthHeuristics) {
+                let max_width = self.max_width.2;
+                let get_width_value = |
+                    was_set: bool,
+                    override_value: usize,
+                    heuristic_value: usize,
+                    config_key: &str,
+                | -> usize {
+                    if !was_set {
+                        return heuristic_value;
+                    }
+                    if override_value > max_width {
+                        panic!("`{}` cannot have a value that exceeds `max_width`", config_key);
```

Comment:
I know based on the discussion from https://github.com/rust-lang/rustfmt/issues/3710#issuecomment-563562999 and https://github.com/rust-lang/rustfmt/issues/3710#issuecomment-563714012 that we wanted to treat this scenario (user specific a value for an individual width config that's greater than `max_width`) as an error, but wasn't sure how loudly we wanted to error.

Reply:
Ideally, the return type should include the error, and the whole code should be refactored so that we can handle the error in the appropriate place. Looking at the code, though, that seems to be a challenge.

We would like to avoid calling `panic!` if possible. Would you replace it with `eprintln!` and round up the value with `max_width`?

Reply:
> the return type should include the error, and the whole code should be refactored so that we can handle the error in the appropriate place. Looking at the code, though, that seems to be a challenge.

I had a similar thought but agree it'd be a big change best handled in a separate PR. Do you think we need an issue to keep track of making that change or can it just be addressed when the broader set of config changes are made (assuming something like #3873 is still planned)

Reply:
I don't think we need a separate issue, we can address it as a part of Config refactoring.